gateway/engine/core/src/main/java/io/apiman/gateway/engine/impl/ApiRequestExecutorImpl.java

```diff
@@ -223,8 +223,23 @@ protected void recordErrorMetrics(Throwable error) {
      */
     @Override
     public void execute() {
-        // Strip apikey
-        stripApiKey();
+
+        //load the api data based on request
+        registry.getApi(request.getApiOrgId(), request.getApiId(), request.getApiVersion(),
+                (IAsyncResult<Api> apiResult) -> {
+                    if (apiResult.isSuccess()) {
+                        api = apiResult.getResult();
+                    } else if (apiResult.isError()) {
+                        resultHandler.handle(AsyncResultImpl.create(apiResult.getError(), IEngineResult.class));
+                    }
+                });
+
+        //check if api disable key are enabled
+        if (!api.getDisableKeysStrip()) {
```

Comment:
Just my opinion, but I would add a new method or change the name of the this, e.g.
```java
if (api.isKeysStrippingEnabled)
```

Reply:
Thanks, I've missed about method name. Done too! ;)